### PR TITLE
Update Spark website for 2.4.3 release

### PR DESCRIPTION
**Pull request by gatorsmile**
This PR is to add the release note and news for Spark 2.4.3 release.

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @cloud-fan @dongjoon-hyun @marmbrus @rxin @mengxr @srowen

**Review by srowen — APPROVED** [review]

File: downloads.md [hunk 1]
```diff
@@ -26,14 +26,14 @@ $(document).ready(function() {
 
 4. Verify this release using the <span id="sparkDownloadVerify"></span> and [project release KEYS](https://www.apache.org/dist/spark/KEYS).
 
-Note that, Spark is pre-built with Scala 2.12 since version 2.4.2. Previous versions are pre-built with Scala 2.11.
+Note that, Spark is pre-built with Scala 2.11 except version 2.4.2, which is pre-built with Scala 2.11.
```
Comment (dongjoon-hyun) [inline downloads.md]:
```
- 2.4.2, which is pre-built with Scala 2.11.
+ 2.4.2, which is pre-built with Scala 2.12.
```

File: js/downloads.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ var packagesV8 = [hadoop2p7, hadoop2p6, hadoopFree, sources];
 // 2.4.2+
 var packagesV9 = [hadoop2p7, hadoop2p6, hadoopFree, scala2p11_hadoopFree, sources];
 
-addRelease("2.4.2", new Date("04/23/2019"), packagesV9, true);
+addRelease("2.4.3", new Date("05/07/2019"), packagesV9, true);
```
Comment (dongjoon-hyun) [inline js/downloads.js:25]: According to the vote result announcement, `05/06` will be correct.
- https://lists.apache.org/thread.html/0efe232724f9696225d62dba0377f6ce39b6317b69493a3de0f9b7a3@%3Cdev.spark.apache.org%3E

File: site/committers.html [hunk 3]
```diff
@@ -162,6 +162,9 @@
       <h5>Latest News</h5>
       <ul class="list-unstyled">
         
+          <li><a href="/news/spark-2-4-3-released.html">Spark 2.4.3 released</a>
+          <span class="small">(May 08, 2019)</span></li>
```
Comment (dongjoon-hyun) [inline site/committers.html:166]: This also doesn't match with the download page, `05/07`, too. Let's fix this to `(May 06, 2019)`, consistently.

File: js/downloads.js (hunk 2 above)
Comment (gatorsmile) [reply js/downloads.js:25]: The vote passed on 05/06, but the release actually happened next day. See the release date of maven repo: https://mvnrepository.com/artifact/org.apache.spark

File: site/committers.html (hunk 3 above)
Comment (gatorsmile) [reply site/committers.html:166]: This is the news post date. That is today.

File: js/downloads.js (hunk 2 above)
Comment (dongjoon-hyun) [reply js/downloads.js:25]: Then, could you document this in the release manager's guide? This seems to be arguable. Previously, the vote result announce date is used.
cc @cloud-fan

**Comment by gatorsmile** [comment]
@dongjoon-hyun Could you help run the command `jekyll serve` and see whether all the links and descriptions are accurate?

**Comment by dongjoon-hyun** [comment]
Sure, @gatorsmile !

**Review by dongjoon-hyun — APPROVED** [review]
With `jekyll serve`, all links look correctly linked, @gatorsmile .

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

File: js/downloads.js (hunk 2 above)
Comment (cloud-fan) [reply js/downloads.js:25]: The release process usually takes days, so I pick the date that passed the vote. It's also fine to pick the date that started the release process.

**Pull request closed without merging**